doc/source/ray-core/examples/batch_training.ipynb

```diff
@@ -612,23 +614,18 @@
     "    read_into_object_store_spread = read_into_object_store.options(\n",
     "        scheduling_strategy=\"SPREAD\"\n",
     "    )\n",
-    "    read_tasks = [read_into_object_store_spread.remote(file) for file in files]\n",
     "\n",
-    "    # Dictionary of references to data batches with file names as keys\n",
-    "    data_batch_refs_by_file = {}\n",
-    "    for file_id, data_batch_refs in enumerate(ray.get(read_tasks)):\n",
-    "        data_batch_refs_by_file[files[file_id]] = data_batch_refs\n",
+    "    # Dictionary of references to read tasks with file names as keys\n",
+    "    read_tasks_by_file = {\n",
+    "        files[file_id]: read_into_object_store_spread.remote(file)\n",
+    "        for file_id, file in enumerate(files)\n",
+    "    }\n",
     "\n",
-    "    for file, data_batch_refs in data_batch_refs_by_file.items():\n",
-    "        for pickup_location_id, data_batch_ref in data_batch_refs:\n",
-    "            # Cast PyArrow scalar to Python if needed.\n",
-    "            try:\n",
-    "                pickup_location_id = pickup_location_id.as_py()\n",
-    "            except Exception:\n",
-    "                pass\n",
+    "    for file, read_task_ref in read_tasks_by_file.items():\n",
+    "        for pickup_location_id_and_data_batch in iter(ray.get(read_task_ref)):\n",
```

Comment:
pickup_location_id_and_data_batch_ref to make it clear that the entire thing is a ObjectRef?